pac4j-core/src/main/java/org/pac4j/core/profile/ProfileHelper.java

```diff
@@ -56,55 +58,69 @@ public static boolean isTypedIdOf(final String id, final Class<? extends UserPro
-                        logger.debug("userProfile built : {}", userProfile);
-                        return userProfile;
-                    } catch (final Exception e) {
-                        logger.error("Cannot build instance", e);
-                    }
-                }
+                final String completeName = determineProfileClassByName(className);
+                return buildUserProfileByClassCompleteName(typedId, attributes, completeName);
             }
+        } catch (final Exception e) {
+            logger.error("Cannot build instance", e);
         }
         return null;
     }
 
+    private static String determineProfileClassByName(final String className) {
+        final String completeName;
+        if ("CasProfile".equals(className) || "CasProxyProfile".equals(className)) {
+            completeName = "org.pac4j.cas.profile." + className;
+        } else if ("HttpTGTProfile".equals(className)) {
+            completeName = "org.pac4j.cas.profile.HttpTGTProfile";
+        } else if ("SAML2Profile".equals(className)) {
+            completeName = "org.pac4j.saml.profile.SAML2Profile";
+        } else if ("HttpProfile".equals(className)) {
+            completeName = "org.pac4j.http.profile.HttpProfile";
+        } else if ("OidcProfile".equals(className)) {
+            completeName = "org.pac4j.oidc.profile.OidcProfile";
+        } else if ("LdapProfile".equals(className)) {
+            completeName = "org.pac4j.ldap.profile.LdapProfile";
+        } else if ("DbProfile".equals(className)) {
+            completeName = "org.pac4j.sql.profile.DbProfile";
+        } else if ("MongoProfile".equals(className)) {
+            completeName = "org.pac4j.mongo.profile.MongoProfile";
+        } else if ("YahooOpenIdProfile".equals(className)) {
+            completeName = "org.pac4j.openid.profile.yahoo.YahooOpenIdProfile";
+        } else if ("GaeUserServiceProfile".equals(className)) {
+            completeName = "org.pac4j.gae.profile.GaeUserServiceProfile";
+        } else if ("StormpathProfile".equals(className)) {
+            completeName = "org.pac4j.stormpath.profile.StormpathProfile";
+        } else if ("JwtProfile".equals(className)) {
+            completeName = "org.pac4j.jwt.profile.JwtProfile";
+        } else {
+            final String packageName = className.substring(0, className.length() - 7).toLowerCase();
+            completeName = "org.pac4j.oauth.profile." + packageName + '.' + className;
+        }
+        return completeName;
+    }
+
+    private static UserProfile buildUserProfileByClassCompleteName(final String typedId, final Map<String, Object> attributes,
+                                                                   final String completeName)
+            throws Exception {
+        @SuppressWarnings("unchecked")
+        final Constructor<? extends UserProfile> constructor = (Constructor<? extends UserProfile>) Class
+                .forName(completeName).getDeclaredConstructor();
```

Comment:
OK. I just tracked that with the following issue: https://github.com/pac4j/pac4j/issues/405